includes/class-amp-enhancements.php

```diff
@@ -29,34 +29,57 @@ function ( $gtag_opt ) {
 				return $gtag_opt;
 			}
 		);
-		add_filter( 'amp_content_sanitizers', [ __CLASS__, 'amp_content_sanitizers' ] );
+
+		// Check if AMP Plugin is in Standard mode.
+		// In Transitional/Reader modes, AMP pages with kept invalid markup get redirected to the non-AMP version, which is not desired.
+		if ( function_exists( 'amp_is_canonical' ) && amp_is_canonical() ) {
+			add_filter( 'amp_validation_error_sanitized', [ __CLASS__, 'amp_validation_error_sanitized' ], 10, 2 );
+			add_filter( 'amp_validation_error_default_sanitized', [ __CLASS__, 'amp_validation_error_sanitized' ], 10, 2 );
+		}
 	}
 
 	/**
 	 * AMP plus mode.
 	 *
-	 * @param  string $context The context for which AMP plus should be assessed.
 	 * @return bool Should AMP plus be applied.
 	 */
-	public static function should_use_amp_plus( $context = null ) {
+	public static function should_use_amp_plus() {
 		$should = false;
-		if ( isset( $_GET['ampplus'] ) && defined( 'NEWSPACK_AMP_PLUS_CONFIG' ) && is_array( NEWSPACK_AMP_PLUS_CONFIG ) && in_array( $context, NEWSPACK_AMP_PLUS_CONFIG ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( self::is_amp_plus_configured() ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 			$should = true;
 		}
-		return apply_filters( 'should_use_amp_plus', $should, $context );
+		// Check if AMP Plugin is in Standard mode.
+		// In Transitional/Reader modes, AMP pages with kept invalid markup get redirected to the non-AMP version, which is not desired.
+		if ( function_exists( 'amp_is_canonical' ) && false === amp_is_canonical() ) {
+			$should = false;
+		}
+		return apply_filters( 'should_use_amp_plus', $should );
 	}
 
 	/**
-	 * Allow certain scripts to be included in AMP pages.
+	 * Is AMP plus mode configured.
 	 *
-	 * @param array $sanitizers The array of sanitizers, 'MyClassName' => [] // array of constructor params for class.
+	 * @return bool Is AMP plus mode configured.
 	 */
-	public static function amp_content_sanitizers( $sanitizers ) {
-		if ( self::should_use_amp_plus( 'gam' ) ) {
-			require_once NEWSPACK_ABSPATH . 'includes/amp-sanitizers/class-amp-sanitizer-gam.php';
-			$sanitizers['AMP_Sanitizer_GAM'] = [];
+	public static function is_amp_plus_configured() {
+		return defined( 'NEWSPACK_AMP_PLUS_ENABLED' );
```

Comment:
Not a blocker, but why not enforce a `true` value for this as with our other environment flags?

Reply:
This should be `false` by default, though. It's still experimental until Google allows invalid AMP in the search carousel.

Reply:
Good call, this way it can be set to `false` too. Done in 8cf319bed320d5811e4e88dc1c3133410cedf061